tools/rewriter/GenCallAsm.cpp

```diff
@@ -446,12 +468,12 @@ static void emit_set_pkru(AsmWriter &aw, uint32_t target_pkey, Arch arch) {
   }
 }
 
-static void emit_free_stack_space(AsmWriter &aw, size_t stack_arg_size, int stack_alignment, Arch arch) {
+static void emit_free_stack_space(AsmWriter &aw, size_t stack_arg_size, size_t stack_arg_padding, int stack_alignment, Arch arch) {
   if (arch == Arch::X86) {
     // Free stack space used for stack args on the target stack
     if (stack_arg_size > 0) {
       add_comment_line(aw, "Free stack space used for stack args");
-      add_asm_line(aw, "addq $"s + std::to_string(stack_arg_size) + ", %rsp");
+      add_asm_line(aw, "addq $"s + std::to_string(stack_arg_size + stack_arg_padding) + ", %rsp");
```

Comment:
Is the padding here doing something other than rounding the arg size up to the passed alignment?

Reply:
No, that's exactly what we're doing. `stack_arg_size` may not be a multiple of 8 now.

Reply:
I guess my question is: if `stack_arg_padding` is just `round_up(stack_arg_size, 8) - stack_arg_size`, why is it a separate argument? Or am I misunderstanding?

Reply:
It doesn't need to be and I actually change this in https://github.com/immunant/IA2-Phase2/commit/3caefe7baa423b4796e6f466dfff8ee4b4ca07ee but that's not landed yet.